packages/fx-core/src/component/deploy/deployTypes.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  AzureAccountProvider,
+  IProgressHandler,
+  LogProvider,
+  M365TokenProvider,
+  TelemetryReporter,
+} from "@microsoft/teamsfx-api";
+
+export type DeployStep = {
+  driver: "scriptDriver" | "azureAppServiceDriver" | "azureFunctionDriver" | "azureStorageDriver";
+  args: DeployTypes;
+};
+
+export type DeployTypes = {
+  src?: string;
+  buildCommand?: string;
+  dist?: string;
+  resourceId?: string;
+  ignoreFile?: string;
+  scriptWorkingDir?: string;
+  scriptPath?: string;
+};
+
+export type DeployStepArgs = {
+  src: string;
+  dist: string;
+  ignoreFile?: string;
+};
+
+export type ScriptArgs = {
+  scriptWorkingDir: string;
+  scriptPath: string;
+};
+
+export type BuildArgs = {
+  src: string;
+  buildCommand: string;
+};
+
+export type DeployContext = {
```

Comment:
Since it is a common context about FxCore, I am not sure whether "FxContext" is a better name

Reply:
I don't know if other components need to use the content of other Contexts, I've only picked some of the content needed for Deploy here. Because I want to try to control the number of parameters to avoid confusion